src/middlewared/middlewared/schema.py

```diff
@@ -613,7 +613,9 @@ def clean(self, value):
         if value is None:
             return value
         if not isinstance(value, int) or isinstance(value, bool):
-            if isinstance(value, str) and value.isdigit():
+            if isinstance(value, str) and value.lstrip('-').isdigit():
```

Comment:
I propose let's do something like the following:
```
with contextlib.suppress(ValueError):
    return int(value)
```
We keep the string check and try to return converted integer like above - if it works great, otherwise it will raise an error.

Current logic does not take into account if user specifies i.e `--123`
It should be raising an exception at that point but instead it will silently convert it to `-123`